iocore/net/quic/QUICStream.h

```diff
@@ -110,42 +82,93 @@ class QUICStream : public VConnection, public QUICFrameGenerator, public QUICTra
   Event *_send_tracked_event(Event *, int, VIO *);
 
   void _write_to_read_vio(QUICOffset offset, const uint8_t *data, uint64_t data_length, bool fin);
-  void _records_rst_stream_frame(QUICEncryptionLevel level, const QUICRstStreamFrame &frame);
-  void _records_stream_frame(QUICEncryptionLevel level, const QUICStreamFrame &frame);
-  void _records_stop_sending_frame(QUICEncryptionLevel level, const QUICStopSendingFrame &frame);
-
-  QUICStreamErrorUPtr _reset_reason        = nullptr;
-  bool _is_reset_sent                      = false;
-  QUICStreamErrorUPtr _stop_sending_reason = nullptr;
-  bool _is_stop_sending_sent               = false;
-
   QUICConnectionInfoProvider *_connection_info = nullptr;
   QUICStreamId _id                             = 0;
   QUICOffset _send_offset                      = 0;
   QUICOffset _reordered_bytes                  = 0;
 
-  QUICRemoteStreamFlowController _remote_flow_controller;
-  QUICLocalStreamFlowController _local_flow_controller;
-  uint64_t _flow_control_buffer_size = 1024;
-
   VIO _read_vio;
   VIO _write_vio;
 
-  QUICTransferProgressProviderVIO _progress_vio = {this->_read_vio};
-
   Event *_read_event  = nullptr;
   Event *_write_event = nullptr;
 
-  bool _is_transfer_complete = false;
-  bool _is_reset_complete    = false;
-
   // Fragments of received STREAM frame (offset is unmatched)
   // TODO: Consider to replace with ts/RbTree.h or other data structure
   QUICIncomingStreamFrameBuffer _received_stream_frame_buffer;
+};
+
+class QUICBidirectionalStream : public QUICStream, public QUICTransferProgressProvider
```

Comment:
Are you going to add `QUICUnidirectionalStream`?

Reply:
Yep, but not only `QUICUnidirectionalStream`, This will split to `QUICSendStream` or `QUICReceiveStream`. But it should be out of this pr.

Reply:
I'm wondering about we can build `QUICBidirectionalStream` with these unidirectional stream and it make code simple.
Anyway, I agree with it's out of scope.